### DISCUSSION TITLE
How can I broadcast message to all of connected websocket clients

### DISCUSSION THREAD

**Discussion by NKDark** [tokio-rs/axum#1335] [Q&A] [answered]
I want to send msg to all of clients when I received data from message queue like rabbitmq or kafka, but I don't know how to implement it.

**Comment by davidpdrsn** [comment, marked as answer]
Have a look at the chat example https://github.com/tokio-rs/axum/blob/main/examples/chat/src/main.rs

**Reply by NKDark** [reply]
I saw the chat example before I asked, but I don't know how to collect senders

**Reply by davidpdrsn** [reply]
Make a channel, spawn a task that forwards everything sent on the channel to the websocket, store the sender half in some shared collection.

Like this https://github.com/tokio-rs/axum/blob/main/examples/chat/src/main.rs#L66

**Reply by NKDark** [reply]
Thanks a lot. I created a global channel to resolve this problem.
